utilities/zookeeper/R/reichlab-formatter.R

```diff
@@ -37,21 +38,47 @@ format_predictions_for_reichlab_submission <- function(predictions_cards,
     msg = "Predictions are somehow missing necessary columns.")
 
   assert_that(all(predictions_cards$signal %in%
-                    c("confirmed_incidence_num", "deaths_incidence_num")),
+                    c("confirmed_incidence_num", "deaths_incidence_num",
+                      "confirmed_admissions_covid_1d")),
               msg = "Predictions are for unsupported signals")
 
-  assert_that(all(predictions_cards$incidence_period == "epiweek"),
-              msg = "Currently only submitting epiweek forecasts")
+  assert_that(all(predictions_cards$incidence_period %in%
+                    c("epiweek", "day")),
+              msg = "Currently only submitting epiweek/daily forecasts")
+
+  pcard_check <- predictions_cards %>%
+    select(.data$signal, .data$incidence_period, .data$ahead) %>%
+    mutate(
+      chk_period = case_when(
+        str_detect(.data$signal, "incidence") ~ .data$incidence_period == "epiweek",
+        str_detect(.data$signal, "admissions") ~ .data$incidence_period == "day",
+        TRUE ~ FALSE),
+      chk_ahead = case_when(
+        .data$signal == "confirmed_incidence_num" ~ .data$ahead %in% 1:8,
+        .data$signal == "deaths_incidence_num" ~ .data$ahead %in% 1:20,
+        .data$signal == "confirmed_admissions_covid_1d" ~ .data$ahead %in% 0:130,
+        TRUE ~ FALSE
+      )
+    )
+  assert_that(all(pcard_check$chk_period),
+              msg = "some signals have incorrect incidence_period")
+  assert_that(all(pcard_check$chk_ahead),
+              msg = "some signals have invalid aheads")
+
 
   case_quants <- c(0.025, 0.100, 0.250, 0.500, 0.750, 0.900, 0.975)
 
   reichlab_quantile <- predictions_cards %>%
-    mutate(target = if_else(.data$signal == "confirmed_incidence_num",
-                            paste0(.data$ahead, " wk ahead inc case"),
-                            paste0(.data$ahead, " wk ahead inc death")),
-           type = "quantile",
-           quantile = round(.data$quantile, 3)) %>%
+    mutate(
+      target = case_when(
+        .data$signal == "confirmed_incidence_num" ~ paste0(.data$ahead, " wk ahead inc case"),
+        .data$signal == "deaths_incidence_num" ~ paste0(.data$ahead, " wk ahead inc death"),
+        TRUE ~ paste0(.data$ahead, " day ahead inc hosp")
+      ),
+      type = "quantile",
+      quantile = round(.data$quantile, 3)) %>%
     filter(.data$signal == "deaths_incidence_num" |
+             .data$signal == "confirmed_admissions_covid_1d" |
              allowed_quantiles(.data$quantile, case_quants)) %>%
```

Comment:
Quantile forecasts are filtered to the accepted quantile levels for cases, but maybe not for deaths and hospitalizations.

Reply:
Additionally, around here, if there is already a point prediction provided, it might not be recognized and another point forecast based on the median might be tacked on.

Reply:
Filed away in #126 and #128.  These are just validation issues, so it's fine to merge this without addressing them.  However, I believe #128 or a similar issue has caused issues in some forecasting scripts, where the baseline model outputting a point prediction when all the other forecasters did not actually broke some ensemble geo-reaggregation code.